### PR TITLE
Legacy api e2e

### PR DESCRIPTION
**Pull request by bowei**
Make the e2e framework use an adapter so it can use both the old (extensions) and new (networking) APIs

### CODE REVIEW

File: pkg/e2e/legacy_test.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package e2e
+
+import (
+	"reflect"
+	"testing"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
```
Comment (rramkumar1) [inline pkg/e2e/legacy_test.go]: Don't need this extra space I think.

File: pkg/e2e/legacy.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	klog.Infof("Delete %s/%s", ns, name)
+	if new {
+		return crud.C.NetworkingV1beta1().Ingresses(ns).Delete(name, &metav1.DeleteOptions{})
+	}
+	klog.Warning("Using legacy API")
+	return crud.C.ExtensionsV1beta1().Ingresses(ns).Delete(name, &metav1.DeleteOptions{})
+}
+
+func (crud *IngressCRUD) supportsNewAPI() (bool, error) {
+	apiList, err := crud.C.Discovery().ServerResourcesForGroupVersion("networking/v1beta1")
```
Comment (rramkumar1) [inline pkg/e2e/legacy.go]: Do you need to return error? I don't see anything but nil returned.

File: cmd/e2e-test/draining_test.go [hunk 3]
```diff
@@ -86,12 +82,18 @@ func TestDraining(t *testing.T) {
 			}
 			t.Logf("Echo service created (%s/%s)", s.Namespace, "service-1")
 
-			if _, err := Framework.Clientset.NetworkingV1beta1().Ingresses(s.Namespace).Create(ing); err != nil {
+			ing := fuzz.NewIngressBuilder("", "ingress-1", "").
+				AddPath("test.com", "/", "service-1", intstr.FromInt(80)).
+				Build()
+			ing.Namespace = s.Namespace
```
Comment (rramkumar1) [inline cmd/e2e-test/draining_test.go]: Could you just set namespace in the builder? Same for other files too.

File: pkg/e2e/legacy.go (hunk 2 above)
Comment (bowei) [reply pkg/e2e/legacy.go]: done

File: cmd/e2e-test/draining_test.go (hunk 3 above)
Comment (bowei) [reply cmd/e2e-test/draining_test.go]: done

**Pull request merged**